pages/understanding-json-schema/reference/non_json_data.md

```diff
@@ -7,84 +7,128 @@ section: docs
 
-The acceptable values are  `quoted-printable`,
-`base16`, `base32`, and `base64`. If not specified, the encoding is the
-same as the containing JSON document.
 
-Without getting into the low-level details of each of these encodings,
-there are really only two options useful for modern usage:
+The `contentEncoding` keyword specifies the encoding used to store the contents, as specified in [RFC 2054, part 6.1](https://tools.ietf.org/html/rfc2045) and [RFC 4648](https://datatracker.ietf.org/doc/html/rfc4648).
 
-- If the content is encoded in the same encoding as the enclosing JSON
-    document (which for practical purposes, is almost always UTF-8),
-    leave `contentEncoding` unspecified, and include the content in a
-    string as-is. This includes text-based content types, such as
-    `text/html` or `application/xml`.
-- If the content is binary data, set `contentEncoding` to `base64` and
-    encode the contents using
-    [Base64](https://tools.ietf.org/html/rfc4648). This would include
-    many image types, such as `image/png` or audio types, such as
-    `audio/mpeg`.
+The acceptable values are:
+- `quoted-printable`
+- `base16` 
+- `base32` 
+- `base64` 
+ 
+If not specified, the encoding is the same as the containing JSON document.
 
-<Keywords label="single: contentSchema single: media; contentSchema" />
+For modern usage, there are two main scenarios:
 
-## contentSchema
-<Star label="New in draft 2019-09" />
+1. **Same encoding as JSON document**: Leave `contentEncoding` unspecified and include the content in a string as-is. This is suitable for text-based content types (e.g., `text/html`, `application/xml`) and assumes UTF-8 encoding in most cases.
+2. **Binary data**: Set `contentEncoding` to `base64` and encode the content using Base64. This is appropriate for binary content types such as images (`image/png`) or audio files (`audio/mpeg`).
 
-Documentation Coming soon
 
-## Examples
+Example:
 
-The following schema indicates the string contains an HTML document,
-encoded using the same encoding as the surrounding document:
+The following schema indicates that a string contains a PNG file and is encoded using Base64:
 
 ```json
 // props { "isSchema": true }
 {
   "type": "string",
-  "contentMediaType": "text/html"
+  "contentEncoding": "base64",
+  "contentMediaType": "image/png"
 }
 ```
 ```json
 // props { "indent": true, "valid": true }
-"<!DOCTYPE html><html xmlns=\"http://www.w3.org/1999/xhtml\"><head></head></html>"
+"iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAYAAADgdz34AAAABmJLR0QA/wD/AP+gvaeTAAAA..."
 ```
-The following schema indicates that a string contains a PNG image, encoded using Base64:
+
+To better understand how `contentEncoding` and `contentMediaType` are applied in practice, let's consider the process of transmitting non-JSON data:
+
+![Role of contentEncoding and contenMediaType keywords in the transmission of non-JSON data](/media-keywords.png)
```

Comment:
```suggestion
![Role of contentEncoding and contenMediaType keywords in the transmission of non-JSON data](/img//media-keywords.png)
```

The image is now being shown. We need to add the image folder to make that work.

Reply:
Does the website support Mermaid diagrams (Example: https://github.com/json-schema-org/json-schema-spec/pull/1544/files)? If so, it might be nicer because it can be styled to match the rest of the site.

Reply:
Thanks for the recommendation. I added a version of the diagram in Mermaid to test if it works on our website.

Reply:
Looks like the website doesn't support mermaid :cry:. That's too bad. Thanks for trying.